airbyte-webapp/src/components/connectorBuilder/Builder/BuilderSidebar.tsx

```diff
@@ -0,0 +1,136 @@
+import { AddStreamButton } from "./AddStreamButton";
+import styles from "./BuilderSidebar.module.scss";
+import { UiYamlToggleButton } from "./UiYamlToggleButton";
+
+export type BuilderView = "global" | number;
+
+interface ViewSelectButtonProps {
+  className?: string;
+  selected: boolean;
+  onClick: () => void;
+}
+
+const ViewSelectButton: React.FC<React.PropsWithChildren<ViewSelectButtonProps>> = ({
+  children,
+  className,
+  selected,
+  onClick,
+}) => {
+  return (
+    <button
+      className={classnames(className, styles.viewButton, {
+        [styles.selectedViewButton]: selected,
+        [styles.unselectedViewButton]: !selected,
+      })}
+      onClick={onClick}
+    >
+      {children}
+    </button>
+  );
+};
+
+interface BuilderSidebarProps {
+  className?: string;
+  toggleYamlEditor: () => void;
+  onViewSelect: (selected: BuilderView, streamName?: string) => void;
+  selectedView: BuilderView;
+}
+
+export const BuilderSidebar: React.FC<BuilderSidebarProps> = ({
+  className,
+  toggleYamlEditor,
+  onViewSelect,
+  selectedView,
+}) => {
+  const { formatMessage } = useIntl();
+  const { openConfirmationModal, closeConfirmationModal } = useConfirmationModalService();
+  const { yamlManifest } = useConnectorBuilderState();
+  const { values, setValues } = useFormikContext<BuilderFormValues>();
+  const handleResetForm = () => {
+    openConfirmationModal({
+      text: "connectorBuilder.resetModal.text",
+      title: "connectorBuilder.resetModal.title",
+      submitButtonText: "connectorBuilder.resetModal.submitButton",
+      onSubmit: () => {
+        setValues(DEFAULT_BUILDER_FORM_VALUES);
+        onViewSelect("global");
+        closeConfirmationModal();
+      },
+    });
+  };
+
+  return (
+    <div className={classnames(className, styles.container)}>
+      <UiYamlToggleButton yamlSelected={false} onClick={toggleYamlEditor} />
+
+      {/* TODO: replace with uploaded img when that functionality is added */}
+      <img
+        className={styles.connectorImg}
+        src="/logo.png"
+        alt={formatMessage({ id: "connectorBuilder.connectorImgAlt" })}
+      />
+
+      <div className={styles.connectorName}>
+        <Heading as="h2" size="sm" className={styles.connectorNameText}>
+          {values.global?.connectorName}
+        </Heading>
+      </div>
+
+      <ViewSelectButton
+        className={styles.globalConfigButton}
+        selected={selectedView === "global"}
+        onClick={() => onViewSelect("global")}
+      >
+        <FontAwesomeIcon icon={faSliders} />
+        <FormattedMessage id="connectorBuilder.globalConfiguration" />
+      </ViewSelectButton>
+
+      <div className={styles.streamsHeader}>
+        <Text className={styles.streamsHeading} size="xs" bold>
+          <FormattedMessage id="connectorBuilder.streamsHeading" values={{ number: values.streams.length }} />
+        </Text>
+
+        <AddStreamButton
+          className={styles.addStreamButton}
```

Comment:
You are adding some styles here and then there are additional styles in the AddStreamButton component itself. Can we consolidate these in a single place?

Reply:
This is a pattern I've been doing for a little while, but let me know if you think I shouldn't be doing it:
- When I create a custom component, I've usually been trying to keep positional styling (e.g. the `addStreamButton` style in BuilderSidebar.module.scss which just sets margin values) in the same `.module.scss` file as the parent component that holds the custom component, because that feels like the most natural place to control the position. 
- I then put the styling code for the internals of the custom component (e.g. the `addButton` style in the AddStreamButton.module.scss file) in the custom component's own scss file, since that feels like the most natural place to style internals of a component.

Basically it felt weird to me to put positional styling inside the custom component's own styles, because if I want to use that custom component in another place in the future then I would need to lift those positional styles back up to the parent anyway. What do you think?

Reply:
> But what do you think of that pattern overall?

Thanks for the explanation, that makes a lot of sense to me as a pattern - Ideally I hope we can abstract away most of these concerns into library components, but I'll bring it up in the chapter sync.